### PR TITLE
New functions `emptyVotingProcedures`, `singletonVotingProcedures` and `mergeVotingProcedures`

### PR DESCRIPTION
**Pull request by newhoggy**
# Changelog

```yaml
- description: |
    New functions `emptyVotingProcedures`, `singletonVotingProcedures` and `mergeVotingProcedures`
# uncomment types applicable to the change:
  type:
  - feature        # introduces a new feature
  # - breaking       # the API has changed in a breaking way
  - compatible     # the API has changed but is non-breaking
  # - optimisation   # measurable performance improvements
  # - improvement    # QoL changes e.g. refactoring
  # - bugfix         # fixes a defect
  # - test           # fixes/modifies tests
  # - maintenance    # not directly related to the code
  # - release        # related to a new release preparation
  # - documentation  # change in code docs, haddocks...
```

# Context

Useful functions for `VotingProcedures`.

For use cases, see: https://github.com/input-output-hk/cardano-cli/pull/203

# Checklist

- [ ] Commit sequence broadly makes sense and commits have useful messages
- [ ] The change log section in the PR description has been filled in
- [ ] New tests are added if needed and existing tests are updated.  These may include:
  - golden tests
  - property tests
  - round trip tests
  - integration tests
  See [Running tests](https://github.com/input-output-hk/cardano-node-wiki/wiki/Running-tests) for more details
- [ ] The version bounds in `.cabal` files are updated
- [ ] CI passes. See note on CI.  The following CI checks are required:
  - [ ] Code is linted with `hlint`.  See `.github/workflows/check-hlint.yml` to get the `hlint` version
  - [ ] Code is formatted with `stylish-haskell`.  See `.github/workflows/stylish-haskell.yml` to get the `stylish-haskell` version
  - [ ] Code builds on Linux, MacOS and Windows for `ghc-8.10.7` and `ghc-9.2.7`
- [ ] The changelog section in the PR is updated to describe the change
- [ ] Self-reviewed the diff

### CODE REVIEW

File: cardano-api/internal/Cardano/Api/Governance/Actions/VotingProcedure.hs [hunk 1]
```diff
@@ -256,3 +256,36 @@ instance IsShelleyBasedEra era => HasTextEnvelope (VotingProcedures era) where
...
+  -> L.GovActionId (L.EraCrypto (ShelleyLedgerEra era))
+  -> L.VotingProcedure (ShelleyLedgerEra era)
+  -> VotingProcedures era
+singletonVotingProcedures _ voter govActionId votingProcedure =
+  VotingProcedures
+    $ L.VotingProcedures
+    $ Map.singleton voter
+    $ Map.singleton govActionId votingProcedure
+
+-- | Right biased merge ov Voting procedures.
```
Comment (carbolymer) [inline cardano-api/internal/Cardano/Api/Governance/Actions/VotingProcedure.hs:281]: typo, should be "of"

**Review by carbolymer — APPROVED** [review]

File: cardano-api/internal/Cardano/Api/Governance/Actions/VotingProcedure.hs (hunk 1 above)
Comment (newhoggy) [reply cardano-api/internal/Cardano/Api/Governance/Actions/VotingProcedure.hs]: Fixed.  Thanks!

File: cardano-api/internal/Cardano/Api/Governance/Actions/VotingProcedure.hs [hunk 2]
```diff
@@ -256,3 +256,36 @@ instance IsShelleyBasedEra era => HasTextEnvelope (VotingProcedures era) where
...
+  -> L.VotingProcedure (ShelleyLedgerEra era)
+  -> VotingProcedures era
+singletonVotingProcedures _ voter govActionId votingProcedure =
+  VotingProcedures
+    $ L.VotingProcedures
+    $ Map.singleton voter
+    $ Map.singleton govActionId votingProcedure
+
+-- | Right biased merge of Voting procedures.
+mergeVotingProcedures :: ()
```
Comment (Jimbo4350) [inline cardano-api/internal/Cardano/Api/Governance/Actions/VotingProcedure.hs:282]: I think we need to be careful here. I would argue something has gone wrong if a voter votes twice (with different votes) on a single proposal and we shouldn't ignore it.

**Review by Jimbo4350 — CHANGES_REQUESTED** [review]
We need to discuss `mergeVotingProcedures`

File: cardano-api/internal/Cardano/Api/Governance/Actions/VotingProcedure.hs (hunk 2 above)
Comment (newhoggy) [reply cardano-api/internal/Cardano/Api/Governance/Actions/VotingProcedure.hs:282]: I agree.  Can we make this a TODO?

**Comment by newhoggy** [comment]
Added `unsafe` prefix and TODO to cleanup merge function later.

**Pull request merged**